File: Java/Pow(x, n).java
public class Solution {
    public double myPow(double x, int n) {
        if(n == 0){//Base case.
            return 1;
        }
        
        boolean isNegative = false;
        if(n < 0){
            n = -n;
            isNegative = true;
        }
        
        double ans = myPow(x, n/2);
        ans = ans * ans;
        if(n % 2 != 0){
           ans = ans * x;
        }
        
        if(isNegative){
            ans = 1/ans;
        }
        return ans;
    }
}

File: C++/Pow(x, n).cpp
class Solution {
public:
    double myPow(double x, int n) {
       if(n == 0) 
            return 1;
       double nBy2 = n/2;
       double power = n > 0 ? myPow(x, nBy2) : myPow(x = 1/x , -nBy2 );
       return    power * power * ( n - 2*nBy2 ? x : 1);
    }
};
